File: Data Structures & Algorithms/string-encode-and-decode/submission-3.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        # We don't need the empty check anymore; an empty list 
        # naturally loops zero times and returns just "#"
        sizes, res = [], ""
        for s in strs:
            sizes.append(len(s))
        for sz in sizes:
            res += str(sz) + ','
        res += '#'
        for s in strs:
            res += s
        return res
    
    def decode(self, s: str) -> List[str]:
        sizes, res, i = [], [], 0
        
        # Read the sizes until we hit the marker
        while s[i] != '#':
            cur = ""
            while s[i] != ',':
                cur += s[i]
                i += 1
            sizes.append(int(cur))
            i += 1 # skip the comma
            
        i += 1 # skip the '#' marker
        
        # Slice out the strings using our saved sizes
        for sz in sizes:
            res.append(s[i : i + sz]) # Fixed the double-nested append!
            i += sz
            
        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        sort(nums.begin(),nums.end());
        int res=0,cur=nums[0],stk=0,i=0;
        while (i<nums.size()){
            if (cur!=nums[i]){
                cur = nums[i];
                stk = 0;
            }
            while(i<nums.size()&&nums[i]==cur){
                i++;
            }
            stk++;
            cur++;
            res = max(res,stk);
        }
        return res;
    }
};
